clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -31,14 +31,22 @@
     public static final int DEFAULT_THROTTLE_TIME = 0;
 
     protected Send toSend(String destination, ResponseHeader header, short apiVersion) {
-        return new NetworkSend(destination, serialize(apiVersion, header));
+        return new NetworkSend(destination, serialize(header.toStruct(), toStruct(apiVersion)));
     }
 
```

Comment:
Do you think it's worth adding a method to return the header version here? Subclasses can access it with `this.api.headerVersion(this.version())`, but it might be nice to have `AbstractResponse#headerVersion`

Reply:
Interesting idea, but let's see if we need it anywhere first.